docs/100_OVERVIEW.md

```diff
@@ -0,0 +1,147 @@
+# OCR2 Automation Design Overview
+The intent of the plugin design is to separate responsibility of components,
+ensure modularity, and provide accessible types for viewing content in 
+messages via external sources (like telemetry).
+
+## Glossary of Terms
+- Observation: two different types; one from the `Plugin` and one from an `Observer`
+- Report: encoded instructions and/or values for on-chain execution
+- BlockKey: a block identifier (typically a big.Int on many chains)
+- UpkeepIdentifier: an identifier for an upkeep (typically a big.Int)
+- UpkeepKey: a combination of `BlockKey` and `UpkeepIdentifier`
+- NetworkConfig: also called off-chain config in OCR; network configuration
+- In-flight: upkeeps are in-flight if they are reported, but not transmitted
+- Transmit: an attempt to include a report on-chain
+
+## Components
+
+### Plugin
+The `Plugin` provides methods for LibOCR to call as a valid OCR2 plugin. The 
+responsibility of each method is defined below. Generally, a `Plugin` gets
+observations from a list of multiple `Observer`, uses a `Merger` to merge
```

Comment:
yet to read it completely, but q at this point is the need for multiple observers instead of a single one